### PR TITLE
DS-850 | Annotated image component

### PR DESCRIPTION
**Pull request by yvonnetangsu**
# READY FOR REVIEW

# Summary
- Add Annotated Image component

# Review By (Date)
- Retro

# Criticality
- 5

# Review Tasks

## Setup tasks and/or behavior to test

1. Go to Momentum/Test/Yvonne Test Annotated Image
2. Select PR #343 from the visual editor
3. Play with all the variants on the page and test all responsive
4. Quick check for accessibility (I've tested with Voice Over and it seems ok)

# Associated Issues and/or People
- JIRA ticket(s) - DS-850

### CODE REVIEW

**Comment by sherakama** [comment]
This is looking good. 

Screenshot 2024-10-11 at 15 38 55
Screenshot 2024-10-11 at 15 39 21
Screenshot 2024-10-11 at 15 39 29

The sizes of the nodes feel nice on my phone and don't seem too crammed together. It will be good to document and let the content authors know that the points can't be too close together or they will overlap on mobile. 

One thing on mobile I noticed was a weird resize. When I click on a button to open the modal, the content gets a gap of content on the right side. Perhaps something to do with flexbox and a new modal div in the dom? The site jumps back to the correct size when the modal is closed.

**Comment by sherakama** [comment]
The content author experience in Storyblok is nice. I like your grouping, field names, and descriptions.

File: components/Storyblok/Storyblok.types.ts [hunk 1]
```diff
@@ -72,3 +74,28 @@ export type SbColorStopProps = {
...
+
+export type SbSliderType = {
+  value?: number;
+}
+
+// Used for Annotated Image component
+export type SbImageHotspotModalContentType =  'text-image' | 'fullwidth-image' | 'text' | 'component';
+export type SbImageHotspotDescriptionSizeType = 'card' | 'default' | 'big';
+
+export type SbImageHotspotType = {
```
Comment (sherakama) [inline components/Storyblok/Storyblok.types.ts:86]: One day we will generate these from Storyblok...

**Comment by yvonnetangsu** [comment]
> The sizes of the nodes feel nice on my phone and don't seem too crammed together. It will be good to document and let the content authors know that the points can't be too close together or they will overlap on mobile.
> 
> One thing on mobile I noticed was a weird resize. When I click on a button to open the modal, the content gets a gap of content on the right side. Perhaps something to do with flexbox and a new modal div in the dom? The site jumps back to the correct size when the modal is closed.

Hey thanks for looking at this @sherakama! I just pushed up another big commit. Need another one to move all my styles in one place then I think I'm done. I don't seem to see the weird resize on my iPhone - I wonder if it's an android thing. It can have something to do with scrollbars. Next time we meet let's do a sharing 😄 

You're probably out soon so I'll likely push this to PROD after another check. Since this is a new feature, I think if you find anything retro next week I can fix it after 🤠 

Happy Canadian Thanksgiving - enjoy some quality family time and pie!! 🦃 🥧

**Comment by sherakama** [comment]
>  I don't seem to see the weird resize on my iPhone - I wonder if it's an android thing.

I was able to reproduce it with my browser and the `iPhone 14 Pro` option. Do you see the layout shift when you open/close the modal in Chrome Devtools?

**Comment by sherakama** [comment]
> You're probably out soon so I'll likely push this to PROD after another check

Sure, I don't see anything that would stop it. I didn't do a very comprehensive A11y review however. I was able to tab navigate and the focus traps/sets were working well as far as I could tell.

You should probably take a pass with voiceover once before going live tho.

**Comment by yvonnetangsu** [comment]
> > I don't seem to see the weird resize on my iPhone - I wonder if it's an android thing.
> 
> I was able to reproduce it with my browser and the `iPhone 14 Pro` option. Do you see the layout shift when you open/close the modal in Chrome Devtools?

Oh I see what you mean. I did see it on DevTools for iPhone but it doesn't happen on the real iPhone

**Comment by yvonnetangsu** [comment]
> > You're probably out soon so I'll likely push this to PROD after another check
> 
> Sure, I don't see anything that would stop it. I didn't do a very comprehensive A11y review however. I was able to tab navigate and the focus traps/sets were working well as far as I could tell.
> 
> You should probably take a pass with voiceover once before going live tho.

Thanks - will do! I'll probably send this over to SODA as well.

**Comment by yvonnetangsu** [comment]
Going to push this live. Tested on VO and iPhone. This won't be published until Nov so I'm not too worried about making it live now. 🤠

**Pull request merged**